app/controllers/access_permissions/projects_controller.rb

```diff
@@ -108,12 +123,42 @@ def set_project
       render_404 unless @project
     end
 
+    def propagate_job(user_assignment)
+      UserAssignments::PropagateAssignmentJob.perform_later(
+        @project,
+        user_assignment.user,
+        user_assignment.user_role,
+        current_user
+      )
+    end
+
     def check_manage_permissions
       render_403 unless can_manage_project_users?(@project)
     end
 
     def check_read_permissions
       render_403 unless can_read_project?(@project)
     end
+
+    def available_users
```

Comment:
could possibly be moved to the assignable concern, as `assignable_users`

Reply:
This logic is dedicated to projects